### PR TITLE
Fix DeviceTwinSample

### PR DESCRIPTION
**Pull request by brycewang-microsoft**
Detail in comments.

### CODE REVIEW

File: service/iot-service-samples/device-twin-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceTwinSample.java [hunk 1]
```diff
@@ -49,22 +49,22 @@ public static void main(String[] args) throws Exception
         {
             // Manage complete twin
             // ============================== get initial twin properties =============================
-            Twin device = getInitialState(twinClient, deviceId);
```
Comment (brycewang-microsoft) [inline service/iot-service-samples/device-twin-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceTwinSample.java:52]: We don't need to cache the device twin and then pass it in each twin operation method below. When we did so, we were using the **existing** `Twin` instance, as a result, a replace call with the cached device twin would also update the unintended properties. We should've been created a new `Twin` instance each time when we do a twin operation.

File: service/iot-service-samples/device-twin-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceTwinSample.java [hunk 2]
```diff
@@ -88,50 +88,57 @@ private static Twin getInitialState(TwinClient twinClient, String deviceId) thro
...
-        System.out.println(deviceTwin);
+        Twin updatedTwin = twinClient.get(deviceId);
+        System.out.println(updatedTwin);
     }
 
-    private static void replaceDesiredProperties(TwinClient twinClient, Twin deviceTwin) throws IOException, IotHubException
+    private static void replaceDesiredProperties(TwinClient twinClient) throws IOException, IotHubException
     {
-        deviceTwin.getDesiredProperties().put("temp", new Random().nextInt(TEMPERATURE_RANGE));
+        Twin newTwins = new Twin(deviceId);
```
Comment (brycewang-microsoft) [inline service/iot-service-samples/device-twin-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceTwinSample.java:107]: This is the new `Twin` object I am talking about above.

File: service/iot-service-samples/device-twin-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceTwinSample.java [hunk 3]
```diff
@@ -88,50 +88,57 @@ private static Twin getInitialState(TwinClient twinClient, String deviceId) thro
...
+        newTwins.getDesiredProperties().put("temp", new Random().nextInt(TEMPERATURE_RANGE));
 
+        // By replacing the twin rather than patching it, any desired properties that existed on the twin prior to this call
```
Comment (brycewang-microsoft) [inline service/iot-service-samples/device-twin-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceTwinSample.java:110]: Get the comments back as the sample is working as expected now.

**Review by timtay-microsoft — APPROVED** [review]

**Review by schoims — APPROVED** [review]

**Review by andyk-ms — APPROVED** [review]

**Review by patilsnr — APPROVED** [review]

**Pull request merged**